### PR TITLE
refactor(i18n): lazy load languages

### PR DESCRIPTION
**Pull request by forehalo**
closes AF-1397

### CODE REVIEW

**Comment by forehalo** [comment]
* **#8456** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8456?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @forehalo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/core/src/modules/i18n/context.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { useService } from '@toeverything/infra';
+import { type PropsWithChildren, useEffect } from 'react';
+
+import { I18nService } from './services/i18n';
+
+export function I18nProvider({ children }: PropsWithChildren) {
+  const i18n = useService(I18nService).i18n;
+
+  useEffect(() => {
+    i18n.init();
```
Comment (EYHN) [inline packages/frontend/core/src/modules/i18n/context.ts]: Should we wait for i18n to be ready before rendering children? there might be flickering from en to other languages?

Comment (forehalo) [reply packages/frontend/core/src/modules/i18n/context.ts]: en better than nothing

File: packages/frontend/i18n/src/index.ts [hunk 2]
```diff
@@ -35,88 +33,87 @@ declare module 'react-i18next' {
...
+        read: (lng: Language, _ns: string, callback) => {
+          const resource = SUPPORTED_LANGUAGES[lng].resource;
+          if (typeof resource === 'function') {
+            resource()
+              .then(data => {
+                logger.info(`Loaded i18n ${lng} resource`);
+                callback(null, data);
+              })
+              .catch(err => {
+                logger.error(`Failed to load i18n ${lng} resource`, err);
```
Comment (pengx17) [inline packages/frontend/i18n/src/index.ts]: fallback to `defaultLng` when load error?

Comment (forehalo) [reply packages/frontend/i18n/src/index.ts]: fallback setting at L67, 'en' is always loaded

Comment (pengx17) [reply packages/frontend/i18n/src/index.ts]: i think `fallbackLng` at `L67` is for individual missing trans keys for that locale.

Comment (forehalo) [reply packages/frontend/i18n/src/index.ts]: the resolver will always traverse all fallbacks to find a key. when error happens, i18next will fill that resources with empty object

```
langs = ['zh-Hans', 'en']

foreach lang in langs
  if lang.resolve(key) 
    break
```

**Review by pengx17 — APPROVED** [review]

**Pull request merged**